indexer-balances/src/db_adapters/balance_changes.rs

```diff
@@ -699,3 +743,62 @@ async fn get_account_view(
         ),
     }
 }
+
+// Feature "rpc-sanity-check" performs an extra call to the RPC (RPC_SANITY_CHECK_URL)
+// compares the results and falls back to the RPC_SANITY_CHECK_URL's result in case of mismatch
+// The corresponding warning log is being emitted.
+#[cfg(feature = "rpc-sanity-check")]
+async fn rpc_sanity_check(
+    sanity_json_rpc_client: &near_jsonrpc_client::JsonRpcClient,
+    account_id: &near_indexer_primitives::types::AccountId,
+    block_hash: &near_indexer_primitives::CryptoHash,
+    account_response: near_jsonrpc_primitives::types::query::RpcQueryResponse,
+) -> near_jsonrpc_primitives::types::query::RpcQueryResponse {
+    let query = near_jsonrpc_client::methods::query::RpcQueryRequest {
+        block_reference: near_primitives::types::BlockReference::BlockId(
+            near_primitives::types::BlockId::Hash(*block_hash),
+        ),
+        request: near_primitives::views::QueryRequest::ViewAccount {
+            account_id: account_id.clone(),
+        },
+    };
+
+    let sanity_check_account_response = match sanity_json_rpc_client.call(query).await {
+        Ok(res) => res,
+        Err(err) => {
+            tracing::error!(
+                target: crate::LOGGING_PREFIX,
+                "RPC sanity check call failed with error \n{:#?}\n
+                Returning the result from RPC_URL. Data sanity might be corrupted!",
```

Comment:
Data sanity of 100x
Please drop the last sentence or add this detail

Reply:
Perhaps you can help me to rephrase it. I was trying to give the reason why we fall back to the reliable data source results because otherwise data sanity might be corrupted.

Agree I have messed up with texting 😬

Reply:
Your phrasing is fine, but we need to clarify which data sanity might be corrupted.
`Returning the result from reliable RPC_URL. Data sanity of 100x project might be corrupted!`